logaccess/loki/src/main/kotlin/LokiLogFileProvider.kt

```diff
@@ -104,7 +104,18 @@ class LokiLogFileProvider(
                 }
 
                 if (statements.size >= config.limit && deDuplicatedStatements.isNotEmpty()) {
-                    downloadChunk(statements.last().timestamp, deDuplicatedStatements)
+                    val lastTimestamp = statements.last().timestamp
+
+                    if (lastTimestamp != from) {
+                        downloadChunk(lastTimestamp, deDuplicatedStatements)
+                    } else {
+                        logger.warn(
+                            "Possible loss of log data for run '{}' and source '{}' " +
+                            "due to number of identical timestamps exceeds chunk size limit.",
+                            ortRunId,
+                            source
+                        )
+                    }
```

Comment:
As a quick fix this is good enough, but I think we also need a proper solution that does not have the risk of losing log data. An inefficient solution could be to reload the chunk with a larger size until the end timestamp does not equal the start timestamp anymore.
But I also wonder why chunking was implemented in the first place instead of just downloading the full logs.